pkg/server/ptypes/project_template.go

```diff
@@ -0,0 +1,117 @@
+package ptypes
+
+import (
+	validation "github.com/go-ozzo/ozzo-validation/v4"
+	"github.com/imdario/mergo"
+	"github.com/mitchellh/go-testing-interface"
+	"github.com/stretchr/testify/require"
+
+	"github.com/hashicorp/waypoint/internal/pkg/validationext"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+)
+
+const (
+	PROJECT_TEMPLATE_ID_LENGTH               = 64
+	PROJECT_TEMPLATE_NAME_LENGTH             = 64
+	PROJECT_TEMPLATE_TAG_LENGTH              = 64
+	PROJECT_TEMPLATE_SUMMARY_LENGTH          = 64
+	PROJECT_TEMPLATE_EXPANDED_SUMMARY_LENGTH = 512
+	PROJECT_TEMPLATE_README_LENGTH           = 1024 ^ 2
+	PROJECT_TEMPLATE_WAYPOINT_HCL_LENGTH     = 1024 ^ 2
+
+	TERRAFORM_NOCODE_MODULE_SOURCE_LENGTH  = 4096
```

Comment:
😮 wow that seems pretty high

Reply:
Good question!

TFC gets the name from github, which empirically has a 100 character limit:

<img width="725" alt="Screenshot 2023-06-20 at 5 45 02 PM" src="https://github.com/hashicorp/waypoint/assets/8404559/63b84fae-32a0-4a06-8dca-1a97d93d196f">

Reply:
Updated to be bigger. I'm setting it above github's limit, because our main concern is around data storage, and if github slightly increases their limit in the future i'd rather avoid having to change it on our side.